blocks/testing/include/gnuradio-4.0/testing/bm_test_helper.hpp

```diff
@@ -47,50 +43,50 @@ class source : public gr::Block<source<T, min, count>> {
         return x;
     }
 
-    gr::work::Result
-    work(std::size_t requested_work) {
-        const std::size_t n_to_publish = _n_samples_max - n_samples_produced;
-        if (n_to_publish > 0) {
-            auto &port   = out;
-            auto &writer = port.streamWriter();
-            if (n_samples_produced == 0) {
-                port.publishTag({ { "N_SAMPLES_MAX", _n_samples_max } }, static_cast<gr::Tag::signed_index_type>(_n_tag_offset)); // shorter version
-            }
-
-            if constexpr (use_bulk_operation) {
-                std::size_t n_write = std::clamp(n_to_publish, 0UL, std::min(writer.available(), port.max_buffer_size()));
-                if (n_write == 0UZ) {
-                    return { requested_work, 0UZ, gr::work::Status::INSUFFICIENT_INPUT_ITEMS };
-                }
-
-                writer.publish( //
-                        [this](std::span<T> output) {
-                            for (auto &val : output) {
-                                val = processOne();
-                            }
-                        },
-                        n_write);
-            } else {
-                auto [data, token] = writer.get(1);
-                if (data.size() == 0UZ) {
-                    return { requested_work, 0UZ, gr::work::Status::ERROR };
-                }
-                data[0] = processOne();
-                writer.publish(token, 1);
-            }
-            return { requested_work, 1UZ, gr::work::Status::OK };
-        } else {
-            return { requested_work, 0UZ, gr::work::Status::DONE };
-        }
-    }
+    // gr::work::Result
```

Comment:
Is this function still needed? It looks obsolete.

Reply:
I tried to minimise the refactoring since this is still largely code that has been introduced by @mattkretz for the compile-time merge and SIMD API. Some of the tests could be rationalised but I do not want to break too many things at the same time.

Reply:
removed.